media/js/app/taxonomy/taxonomy.js

```diff
@@ -442,31 +448,45 @@
 
             // split the url.
             var urls = value.split(',');
+            var flagArray = [];
```

Comment:
Hmmm, I'd probably create two helper functions -- isJson(url), isSkos(url) that returns true or false based on the regex match. Then, at Ln. 462, if (!isJson(urls[i]) && !isSkos(urls[i]) { show error }. At line  477,  this.getTheOnomy(urls[j], this.selected, isJson(urls[j]); The helper function can also be used in refreshOnomy to maintain consistency rather than the quick (urls[i].indexOf("json") > -1) check.